RecoMuon/TrackerSeedGenerator/plugins/TSGForOI.cc

```diff
@@ -1,425 +1,475 @@
-  desc.add<std::string>("propagatorName","PropagatorWithMaterial");
-  descriptions.add("TSGForOI",desc);
-}
-
-DEFINE_FWK_MODULE(TSGForOI);
+/**
+  \class    TSGForOI
+  \brief    Create L3MuonTrajectorySeeds from L2 Muons updated at vertex in an outside in manner
+  \author   Benjamin Radburn-Smith, Santiago Folgueras
+ */
+
+#include "RecoMuon/TrackerSeedGenerator/plugins/TSGForOI.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "DataFormats/Math/interface/deltaR.h"
+
+#include <memory>
+
+using namespace edm;
+using namespace std;
+
+TSGForOI::TSGForOI(const edm::ParameterSet & iConfig) :
+  src_(consumes<reco::TrackCollection>(iConfig.getParameter<edm::InputTag>("src"))),
+  numOfMaxSeedsParam_(iConfig.getParameter<uint32_t>("maxSeeds")),
+  numOfLayersToTry_(iConfig.getParameter<int32_t>("layersToTry")),
+  numOfHitsToTry_(iConfig.getParameter<int32_t>("hitsToTry")),
+  fixedErrorRescalingForHits_(iConfig.getParameter<double>("fixedErrorRescaleFactorForHits")),
+  fixedErrorRescalingForHitless_(iConfig.getParameter<double>("fixedErrorRescaleFactorForHitless")),
+  adjustErrorsDynamicallyForHits_(iConfig.getParameter<bool>("adjustErrorsDynamicallyForHits")),
+  adjustErrorsDynamicallyForHitless_(iConfig.getParameter<bool>("adjustErrorsDynamicallyForHitless")),
+  estimatorName_(iConfig.getParameter<std::string>("estimator")),
+  minEtaForTEC_(iConfig.getParameter<double>("minEtaForTEC")),
+  maxEtaForTOB_(iConfig.getParameter<double>("maxEtaForTOB")),
+  useHitLessSeeds_(iConfig.getParameter<bool>("UseHitLessSeeds")),
+  useStereoLayersInTEC_(iConfig.getParameter<bool>("UseStereoLayersInTEC")),
+  updator_(new KFUpdator()), measurementTrackerTag_(consumes<MeasurementTrackerEvent>(iConfig.getParameter<edm::InputTag>("MeasurementTrackerEvent"))),
+  pT1_(iConfig.getParameter<double>("pT1")),
+  pT2_(iConfig.getParameter<double>("pT2")),
+  pT3_(iConfig.getParameter<double>("pT3")),
+  eta1_(iConfig.getParameter<double>("eta1")),
+  eta2_(iConfig.getParameter<double>("eta2")),
+  eta3_(iConfig.getParameter<double>("eta3")),
+  eta4_(iConfig.getParameter<double>("eta4")),
+  eta5_(iConfig.getParameter<double>("eta5")),
+  eta6_(iConfig.getParameter<double>("eta6")),
+  SF1_(iConfig.getParameter<double>("SF1")),
+  SF2_(iConfig.getParameter<double>("SF2")),
+  SF3_(iConfig.getParameter<double>("SF3")),
+  SF4_(iConfig.getParameter<double>("SF4")),
+  SF5_(iConfig.getParameter<double>("SF5")),
+  SF6_(iConfig.getParameter<double>("SF6")),
+  tsosDiff1_(iConfig.getParameter<double>("tsosDiff1")),
+  tsosDiff2_(iConfig.getParameter<double>("tsosDiff2")),
+  propagatorName_(iConfig.getParameter<std::string>("propagatorName")), 
+  theCategory(string("Muon|RecoMuon|TSGForOI"))
+{
+  produces<std::vector<TrajectorySeed> >();
+}
+
+
+TSGForOI::~TSGForOI() {
+
+}
+
+
+//
+// Produce seeds
+//
+void TSGForOI::produce(edm::StreamID sid, edm::Event& iEvent, const edm::EventSetup& iSetup) const {
+    
+  using namespace std;

```

Comment:
Not needed
(You can probably also avoid the one at line 9, simply checking if an explicit ```std::``` is missing somewhere - not at a first glance...)